src/hera/workflows/_meta_mixins.py

```diff
@@ -474,7 +475,10 @@ def _add_type_hints(
 ) -> Callable[
     ...,
     Callable[
-        PydanticKwargs,  # this adds type hints to the underlying *library* function kwargs
+        Concatenate[
+            object,  # this is the `self` parameter
```

Comment:
I tried to capture the correct class type from the decorated function, but couldn't find a way to do it that didn't break both mypy and pyright. I verified that both mypy and pyright are happy with `object`.